.github/workflows/meta-package.yml

```diff
@@ -38,18 +38,34 @@ jobs:
               github.rest.repos.listTags({
                 owner: context.repo.owner,
                 repo: PACKAGE.substring(PACKAGE.indexOf('/') + 1),
-                per_page: 12,
+                per_page: 50,
               }),
             ])
-            return upstreamTags
-              .filter((tag) => !currrentTags.find((t) => t.name === tag.name))
-              .map((tag) => tag.name)
+            const targetTags = await Promise.allSettled(
+              upstreamTags
+                .filter((tag) => !currrentTags.find((t) => t.name === tag.name))
+                .map((tag) =>
+                  github.rest.git.getRef({
+                    owner: context.repo.owner,
+                    repo: META.substring(META.indexOf('/') + 1),
+                    ref: 'tags/' + tag.name,
+                  })
+                )
+            )
+            return targetTags
+              .filter((req) => req.status === 'rejected') // Non existent tags
+              .map((res) =>
+                res.reason.request.url.substring(
+                  res.reason.request.url.indexOf('%2F') + 3 // 'refs/tags%2Fx.y.z'
```

Comment:
Retrieve the tag from the request.
Definitely ugly.

Reply:
🤔 what about using the tag we already have above? We just need to wrap that promise above somehow?

This might work:

```js
const targetTags = await Promise.allSettled(
  upstreamTags
    .filter((tag) => !currrentTags.find((t) => t.name === tag.name))
    .map((tag) =>
      github.rest.git.getRef({
        owner: context.repo.owner,
        repo: META.substring(META.indexOf('/') + 1),
        ref: 'tags/' + tag.name,
      }).then(res => ({ tag: tag, res }))
    )
)
return targetTags
  .filter(({res}) => res.reason.status === 404) // Non existent tags
  .map({tag} => tag.name)
```

Reply:
That's smart, not that obvious though:
* The `then` is not sufficient as we need to cover the failure as well (actually the failure is the value here)
* `Promise.allSettled` has a weird behavior to always wrap the results into `.value`

```js
const targetTags = await Promise.allSettled(
  upstreamTags
    .filter((tag) => !currrentTags.find((t) => t.name === tag.name))
    .map((tag) =>
      github.rest.git
        .getRef({
          owner: context.repo.owner,
          repo: META.substring(META.indexOf('/') + 1),
          ref: 'tags/' + tag.name,
        })
        .then(() => {})
        .catch((res) => ({ tag: tag.name, status: res.status }))
    )
)
return targetTags
  .filter(({ value }) => value?.status === 404)
  .map(({ value }) => value.tag)

```

Reply:
But that's still a bit wtf 😁

Reply:
From my brief check, octokit never rejects a promise. It's always fulfilled as a success just with an error message in it. Did you see otherwise?

Reply:
Octokit rejects promises if fetch/xhr is failing, but inside a `Promise.allSettled` with `then`/`catch` it makes it wrapped into another always fulfilled promise.